hubblestack/extmods/fdg/cert_discovery.py

```diff
@@ -0,0 +1,168 @@
+# -*- encoding: utf-8 -*-
+"""
+Flexible Data Gathering: cert_discovery
+=============================
+Intention -
+This fdg module allows connecting to open ports on a system and retrieving
+certificate details that might be attached on those ports.
+
+Testing -
+    1. Configure this module through Hubble's schedule using the following configuration
+        fdg_cert_discovery:
+        function: fdg.fdg
+        seconds: __seconds__
+        splay: __splay__
+        returner: __returner__
+        run_on_start: __run_on_start__
+        args:
+          - <path to fdg profile>
+    Make sure that cert_discovery.fdg profile exists and contains Osquery as the first
+    module and this module as the second module.
+    2. Alternately, execute hubble fdg.fdg <path to fdg profile> to run this module via cmd.
+"""
+import OpenSSL
+import ssl
+from socket import setdefaulttimeout
+import logging
+from datetime import datetime
+log = logging.getLogger(__name__)
+
+setdefaulttimeout(3)
+
+def get_certificate_san(x509cert):
+    san = ''
+    ext_count = x509cert.get_extension_count()
+    for i in range(0, ext_count):
+        ext = x509cert.get_extension(i)
+        if 'subjectAltName' in str(ext.get_short_name()):
+            san = ext.__str__()
+    san = san.split(',')
+    return san
+
+def load_certificate(ip, port):
+    """
+    fetch server certificate details and return Json with the first value being the
+    status of ssl.get_server_certificate function and second value being the actual
+    certificate data.
+    """
+    try:
+        log.info("FDG's cert_discovery is checking for ssl cert on {0}:{1}".format(ip,port))
```

Comment:
I guess we can change this to `log.debug` to reduce noise in logs as you are already logging certificate found message in case of a valid cert.

Reply:
done